src/Model/PhasorDynamics/SystemModel.hpp

```diff
@@ -487,39 +507,84 @@ namespace GridKit
         }
       }
 
+      /**
+       * @brief Add bus
+       *
+       */
       void addBus(bus_type* bus)
       {
+        IdxT gridkit_bus_id                  = static_cast<IdxT>(buses_.size());
+        gridkit_bus_indices_[gridkit_bus_id] = bus->busID();
```

Comment:
It is unclear to me what this function is doing. It would be good to explain this better in doxygen comments. For example, something like "Add bus at the end of the bus array and map bus ID with GridKit's ID for the bus."